runelite-client/src/main/java/net/runelite/client/plugins/chatrecolor/ChatRecolorPlugin.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (c) 2018, Hydrox6 <ikada@protonmail.ch>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.chatrecolor;
+
+import com.google.common.base.Strings;
+import com.google.common.eventbus.Subscribe;
+import java.awt.Color;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.MessageNode;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.ResizeableChanged;
+import net.runelite.api.events.SetMessage;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.chat.ChatColor;
+import net.runelite.client.chat.ChatColorType;
+import net.runelite.client.chat.ChatMessageBuilder;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.config.ChatColorConfig;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import javax.inject.Inject;
+
+
+@PluginDescriptor(
+	name = "Chat Text Recolor",
+	enabledByDefault = false
+)
+public class ChatRecolorPlugin extends Plugin
```

Comment:
Hmm, what about making this part of the ChatManager? And add the subscriptions and logic there? The defaults for this plugin should be the correct in-game colors anyway, so it should be fine not having possibility to disable. Or even, add some bool to the ChatColorConfig for disabling it entirely

Reply:
Private Chat colours aren't correct, as there are two options depending on if the player has Split Private chat on in the Runescape options.

With it on, the colour is #00FFFF
Without it on, the colour is #7F0000

Reply:
Perhaps that setting is a playerVar. You could choose the right colour based on that

Reply:
I didn't like it recoloring everything because I didn't trust the defaults. It supports having no default, but it makes the code ugly having to add null checks on all of the colors.

Reply:
You can just then add the boolean there. Currently there is problem that the config is in core, recoloring is in core, and then there is this plugin what hooks them together. And then there are other plugins that can modify the colors too. I would then just add some bool to the new color config to enable it, remove the public methods for messing witht he ChatManager and just do it from inside.